nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/syslog/Syslog5424Reader.java

```diff
@@ -0,0 +1,151 @@
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>(1);
+        properties.add(CHARSET);
+        return properties;
+    }
+
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) {
+        final String charsetName = context.getProperty(CHARSET).getValue();
+        parser = new StrictSyslog5424Parser(Charset.forName(charsetName), NilHandlingPolicy.NULL, NifiStructuredDataPolicy.MAP_OF_MAPS, new SimpleKeyProvider());
+        recordSchema = createRecordSchema();
+    }
+
+    @Override
+    protected List<AllowableValue> getSchemaAccessStrategyValues() {
+        final List<AllowableValue> allowableValues = new ArrayList<>();
+        allowableValues.add(RFC_5424_SCHEMA);
+        return allowableValues;
+    }
+
+    @Override
+    protected AllowableValue getDefaultSchemaAccessStrategy() {
+        return RFC_5424_SCHEMA;
+    }
+
+    @Override
+    protected SchemaAccessStrategy getSchemaAccessStrategy(final String strategy, final SchemaRegistry schemaRegistry, final ConfigurationContext context) {
+        return createAccessStrategy();
+    }
+
+    @Override
+    protected SchemaAccessStrategy getSchemaAccessStrategy(final String strategy, final SchemaRegistry schemaRegistry, final ValidationContext context) {
+        return createAccessStrategy();
+    }
+
+    static RecordSchema createRecordSchema() {
+        final List<RecordField> fields = new ArrayList<>();
+        fields.add(new RecordField(SyslogAttributes.PRIORITY.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(SyslogAttributes.SEVERITY.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(SyslogAttributes.FACILITY.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(SyslogAttributes.VERSION.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(SyslogAttributes.TIMESTAMP.key(), RecordFieldType.TIMESTAMP.getDataType(), true));
+        fields.add(new RecordField(SyslogAttributes.HOSTNAME.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(SyslogAttributes.BODY.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(Syslog5424Attributes.APP_NAME.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(Syslog5424Attributes.PROCID.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(Syslog5424Attributes.MESSAGEID.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(Syslog5424Attributes.STRUCTURED_BASE.key(),
+                RecordFieldType.MAP.getMapDataType(RecordFieldType.MAP.getMapDataType(RecordFieldType.STRING.getDataType()))));
+
+        final RecordSchema schema = new SimpleRecordSchema(fields);
```

Comment:
I know the Grok reader also doesn't set a name, but since most writers will have the default "Schema Write Strategy" as "Set schema.name attribute", I think setting a default name here might be helpful, maybe something like "nifi-syslog-5424".

My first attempt at testing this I created a ConvertRecord processor with a syslog reader and JSON writer using mostly defaults, and I got the error about the writer being unable to write the schema.name attribute because the schema has no name. Once I changed the writer to "Do Not Write Schema" then everything was fine, but I'm figuring some users may not realize what needs to be done.

Reply:
What is the right way to set the name?

Reply:
Sorry for the delay, something like...

SchemaIdentifier schemaIdentifier = StandardSchemaIdentifier.Builder.name("my-name").build();

RecordSchema schema = new SimpleRecordSchema(fields, schemaIdentifier)

Reply:
Ok, done, I think :)